integration/tests/existing_vpc/existing_vpc_test.go

```diff
@@ -0,0 +1,207 @@
+	})
+
+	AfterSuite(func() {
+		// deleteStack(stackName, ctl)
+		Expect(os.RemoveAll(configFile.Name())).To(Succeed())
+	})
+
+	It("supports creating managed and unmanaged nodegroups in the existing VPC", func() {
+		cfg.NodeGroups = []*api.NodeGroup{{
+			NodeGroupBase: &api.NodeGroupBase{
+				Name: ng1,
+			}},
+		}
+		cfg.ManagedNodeGroups = []*api.ManagedNodeGroup{{
+			NodeGroupBase: &api.NodeGroupBase{
+				Name: mng1,
+			}},
+		}
+		// write config file so that the nodegroup creates have access to the vpc spec
+		configData, err := json.Marshal(&cfg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ioutil.WriteFile(configFile.Name(), configData, 0755)).To(Succeed())
+		cmd := params.EksctlCreateCmd.
+			WithArgs(
+				"nodegroup",
+				"--config-file", configFile.Name(),
+				"--verbose", "2",
+			).WithoutArg("--region", params.Region)
+		Expect(cmd).To(RunSuccessfully())
+
+		//GetCluster shows us what subnets/vpc the EKS cluster was created with
+		cmd = params.EksctlGetCmd.WithArgs("cluster", "--name", params.ClusterName, "-o", "yaml")
+		Expect(cmd).To(RunSuccessfullyWithOutputStringLines(
+			ContainElement(ContainSubstring(cfg.VPC.ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["public1"].ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["public2"].ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["public3"].ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["private4"].ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["private5"].ID)),
+			ContainElement(ContainSubstring(cfg.VPC.Subnets.Public["private6"].ID)),
+		))
+	})
+})
+
+func createVPC(stackName string, ctl api.ClusterProvider) *api.ClusterVPC {
+	publicSubnets, privateSubnets, vpcID, securityGroup := createStackAndGetOutputs(stackName, ctl)
+
+	newVPC := api.NewClusterVPC()
+	newVPC.ID = vpcID
+	newVPC.SecurityGroup = securityGroup
+	newVPC.Subnets = &api.ClusterSubnets{
+		Public: api.AZSubnetMapping{
+			"public1": api.AZSubnetSpec{
+				ID: publicSubnets[0],
+			},
+			"public2": api.AZSubnetSpec{
+				ID: publicSubnets[1],
+			},
+			"public3": api.AZSubnetSpec{
+				ID: publicSubnets[2],
+			},
+		},
+		Private: api.AZSubnetMapping{
+			"private4": api.AZSubnetSpec{
+				ID: privateSubnets[0],
+			},
+			"private5": api.AZSubnetSpec{
+				ID: privateSubnets[1],
+			},
+			"private6": api.AZSubnetSpec{
+				ID: privateSubnets[2],
+			},
+		},
+	}
+
+	return newVPC
+}
+
+func createStackAndGetOutputs(stackName string, ctl api.ClusterProvider) ([]string, []string, string, string) {
+	templateBody, err := ioutil.ReadFile("cf-template.yaml")
+	Expect(err).NotTo(HaveOccurred())
+	createStackInput := &cfn.CreateStackInput{
+		StackName: &stackName,
+	}
+	createStackInput.SetTemplateBody(string(templateBody))
+	createStackInput.SetCapabilities(aws.StringSlice([]string{cfn.CapabilityCapabilityIam}))
+	createStackInput.SetCapabilities(aws.StringSlice([]string{cfn.CapabilityCapabilityNamedIam}))
+
```

Comment:
this was a bit hard to read - just to confirm, `createStackAndGetOutputs` creates an existing VPC by creating the stack in `cf-template.yaml`, then it reads the outputs to return the IDs of the public and private subnets, VPC, and SG? then, after this func is called in `createVPC`, these IDs are set on the `newVPC` struct, so that the new cfg can use the VPC ID with `cfg.VPC`. yes?

Reply:
maybe it's worth adding some `By`s to document the process?

Reply:
I just copied/pasted this from the existing, but I'll take a look at improving :smile:

Reply:
> this was a bit hard to read - just to confirm, `createStackAndGetOutputs` creates an existing VPC by creating the stack in `cf-template.yaml`, then it reads the outputs to return the IDs of the public and private subnets, VPC, and SG? then, after this func is called in `createVPC`, these IDs are set on the `newVPC` struct, so that the new cfg can use the VPC ID with `cfg.VPC`. yes?

yes!

Reply:
added some descriptors